emukit/quadrature/acquisitions/squared_correlation.py

```diff
@@ -90,19 +90,15 @@ def _value_terms(self, x: np.ndarray) -> Tuple[np.float, np.ndarray, np.ndarray]
         predictive_cov = np.transpose(qKx - np.dot(qKX, self._graminv_Kx(x)))
         return integral_current_var, y_predictive_var, predictive_cov
 
-    def _gradient_terms(self, x):
+    def _gradient_terms(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
         """
         Computes the terms needed for the gradient of the squared correlation
 
         :param x: (n_points x input_dim) locations where to evaluate
         :return: the gradient of (y_predictive_var, predictive_cov) wrt. x at param x, shapes (n_points, input_dim)
         """
         # gradient of predictive variance of y
-        dvar_dx = self.model.base_gp.kern.dKdiag_dx(x)
-        dKxX_dx1 = self.model.base_gp.kern.dK_dx1(x, self.model.X)
-        graminv_KXx = self._graminv_Kx(x)
-
-        d_y_predictive_var_dx = dvar_dx - 2. * (dKxX_dx1 * np.transpose(graminv_KXx)).sum(axis=2, keepdims=False)
+        d_y_predictive_var_dx = self.model.get_prediction_gradients(x)[1].T
```

Comment:
Good point, it would have probably been more neat if I had implemented get_prediction_gradients method before in VanillaBQ model. However, note that even with the code change SquareCorrelation still can only be used with VanillaBQ, so functionality wise nothing is different. 

I added the gradient functionality to the model in this PR such that the new acquisition UncertaintySampling can be used with any differentiable model. Then I replaced the gradient code in SquareCorrelation where applicable just to make it more readable and not to replicate code.